resources/lang/en/forum.php

```diff
@@ -127,6 +127,49 @@
             'previous' => 'go back 10 posts',
         ],
 
+        'logs' => [
+            '_' => 'Topic logs',
+            'title' => 'topic logs',
+
+            'columns' => [
+                'action' => 'Action',
+                'date' => 'Date',
+                'ip' => 'IP Address',
+                'user' => 'User',
+            ],
+
+            'data' => [
+                'add_tag' => 'added ":data" tag',
+                'announcement' => 'pinned topic and marked as announcement',
+                'delete_post' => 'from topic :data',
+                'edit_topic' => 'to :data',
+                'edit_poll' => 'for :data',
+                'move' => 'from :data',
+                'pin' => 'pinned topic',
+                'post_edited' => 'posted by :data',
+                'remove_tag' => 'removed ":data" tag',
+                'restore_post' => 'from topic :data',
+                'unpin' => 'unpinned topic',
+            ],
+
+            'no_results' => 'no logs found...',
+
+            'operations' => [
```

Comment:
There are a few other operations not supported in osu-web but exist from the old system

Here's the full list of operations in the table

```
+------------------------+
| log_operation          |
+------------------------+
| LOG_DELETE_POST        |
| LOG_POST_EDITED        |
| LOG_MOVE               |
| LOG_LOCK               |
| LOG_MERGE              |
| LOG_USER_LOCK          |
| LOG_SPLIT_DESTINATION  |
| LOG_SPLIT_SOURCE       |
| LOG_UNLOCK             |
| LOG_DELETE_TOPIC       |
| LOG_TOPIC_TYPE_CHANGED |
| LOG_USER_UNLOCK        |
| LOG_FORK               |
| LOG_PIN                |
| LOG_UNPIN              |
| LOG_ISSUE_TAG          |
| LOG_RESTORE_POST       |
| LOG_TOPIC_TYPE         |
| LOG_EDIT_TOPIC         |
| LOG_EDIT_POLL          |
| LOG_RESTORE_TOPIC      |
+------------------------+
```

Reply:
is there any documentation on those not used by osu-web? (see: LOG_MERGE, LOG_USER_LOCK, LOG_SPLIT_DESTINATION, LOG_SPLIT_SOURCE, LOG_TOPIC_TYPE_CHANGED, LOG_USER_UNLOCK, LOG_FORK), how is ``log_data`` stored on them etc.

Reply:
not "documentation" but english texts with log_data placeholders are at <https://github.com/phpbb/phpbb/blob/master/phpBB/language/en/acp/common.php>. if it's not clear what the log_data is from that, searching for the operation in this repo will probably find u the relevant `log::add` call

some of those might be specific to osu though.

Reply:
ok, from delving into the phpbb code i figured out that pretty much all of them (excluding fork) store topic title only, didn't find anything regarding ``LOG_PIN`` and ``LOG_UNPIN`` though, guess it's safe to assume they are just pin and unpin without further information and just display them with no description 🤔